File: TwoSum.cpp
class Solution {
public:
    // ~O(n) in time (unordered_map)
    vector<int> twoSum(vector<int> &numbers, int target) {
        unordered_map<int, int> m;
        for (int i=0; i<numbers.size(); ++i) {
            m[numbers[i]] = i;
        }

        for (int i=0; i<numbers.size(); ++i) {
            int j = m[target-numbers[i]];
            if (j > 0 && j != i) {
                vector<int> ret;
                ret.push_back(i+1);
                ret.push_back(j+1);
                return ret;
            }
        }
    }
};

File: javascript/TwoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    dict = {}
    for (i = 0; i < nums.length; ++i) {
        if (target - nums[i] in dict) {
            return [dict[target - nums[i]], i]
        }
        dict[nums[i]] = i
    }
};
